### PR TITLE
Add CUDA11 build and test

### PR DESCRIPTION
**Pull request by zasdfgbnm**

### CODE REVIEW

File: c10/cuda/CUDAStream.h [hunk 1]
```diff
@@ -155,10 +155,11 @@ class C10_CUDA_API CUDAStream {
...
     #ifndef __HIP_PLATFORM_HCC__
-      int least_priority, greatest_priority;
-      C10_CUDA_CHECK(
-        cudaDeviceGetStreamPriorityRange(&least_priority, &greatest_priority));
-      return std::make_tuple(least_priority, greatest_priority);
+      // Note: this returns the range of priority **supported by PyTorch**, not
+      // the range of priority **supported by CUDA**. The former is a subset of
+      // the latter. Curently PyTorch only supports 0 and -1, which are "low" and
+      // "high" priority.
+      return std::make_tuple(0, -1);
```
Comment (zasdfgbnm) [inline c10/cuda/CUDAStream.h]: cc: @ngimel @ptrblck

Comment (zasdfgbnm) [reply c10/cuda/CUDAStream.h]: Without this change, the test will fail with
```
Jun 26 22:34:46 ======================================================================
Jun 26 22:34:46 FAIL [0.030s]: test_streams_priority (__main__.TestCuda)
Jun 26 22:34:46 ----------------------------------------------------------------------
Jun 26 22:34:46 Traceback (most recent call last):
Jun 26 22:34:46   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_utils.py", line 777, in wrapper
Jun 26 22:34:46     method(*args, **kwargs)
Jun 26 22:34:46   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_utils.py", line 777, in wrapper
Jun 26 22:34:46     method(*args, **kwargs)
Jun 26 22:34:46   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_utils.py", line 387, in wrapper
Jun 26 22:34:46     fn(*args, **kwargs)
Jun 26 22:34:46   File "test_cuda.py", line 946, in test_streams_priority
Jun 26 22:34:46     self.assertEqual(high, s1.priority)
Jun 26 22:34:46   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_utils.py", line 1122, in assertEqual
Jun 26 22:34:46     self.assertTrue(result, msg=msg)
Jun 26 22:34:46 AssertionError: False is not true : Scalars failed to compare as equal! Comparing -2 and -1 gives a difference of 1, but the allowed difference with rtol=0 and atol=0 is only 0!
Jun 26 22:34:46 
Jun 26 22:34:46 ----------------------------------------------------------------------
```

**Review by malfet — APPROVED** [review]

File: c10/cuda/CUDAStream.h (hunk 1 above)
Comment (malfet) [reply c10/cuda/CUDAStream.h:162]: @zasdfgbnm hmm, perhaps we have to modify the test rather than CUDAStream.h

Comment (ngimel) [reply c10/cuda/CUDAStream.h:162]: Perhaps we should leave the API call here and assert that least priority is greater or equal to 0, and greatest priority is less or equal to -1, then creating the tuple with hardcoded values is safe. Having just 2 possible priorities is hardcoded deep in the pytorch.

Comment (zasdfgbnm) [reply c10/cuda/CUDAStream.h:167]: fixed

**Review by malfet — APPROVED** [review]

**Comment by Flamefire** [comment]
Will this be included in the 1.6.x releases? I'm seeing the failure mentioned in https://github.com/pytorch/pytorch/pull/40452#discussion_r446451164 in the rc2 on our machines

**Comment by ngimel** [comment]
1.6 won't include cuda 11 builds.

**Comment by Flamefire** [comment]
At least the fix to c10/cuda/CUDAStream.h  would be great as that does make the tests pass which previously failed on our HPC cluster which is running CUDA 11

**Comment by Flamefire** [comment]
Wait. I was wrong: Our cluster is running CUDA 10.1 and I'm still seeing the issue

**Comment by malfet** [comment]
Added PR for 1.6.1 https://github.com/pytorch/pytorch/pull/41543

File: aten/src/ATen/cudnn/Handle.cpp [hunk 2]
```diff
@@ -16,10 +16,15 @@ void destroyCuDNNHandle(cudnnHandle_t handle) {
...
-#ifdef NO_CUDNN_DESTROY_HANDLE
-#else
-    cudnnDestroy(handle);
-#endif
+//
+// Further note: this is now disabled globally, because we are seeing
+// the same issue as mentioned above in CUDA 11 CI.
+//   - @zasdfgbnm
+//
+// #ifdef NO_CUDNN_DESTROY_HANDLE
```
Comment (gchanan) [inline aten/src/ATen/cudnn/Handle.cpp:24]: do we think this is due to a CUDA 11 bug?

Comment (ngimel) [reply aten/src/ATen/cudnn/Handle.cpp:24]: No, according to the note above errors with out-of-order handle destruction have been present for a long time.

**Pull request closed without merging**